Seminar03/practikum_03.cpp

```diff
@@ -0,0 +1,514 @@
+
+#include <iostream>
+using namespace std;
+int main() {
+
+	//zad1
+	  /*int N;
+	  cin >> N;
+	  int factorial = 1;
+
+	  if(N > 0){
+
+	  for (int i = 1; i <= N; i++) {
+		  factorial *= i;
+	    }
+
+        cout << "Factorial: " << factorial;
+	  } else{
+
+	  cout<< "Invalid input.";
+	  }
+	*/
+
+	  //zad2
+	  /*int n, m;
+	  cout << "Start: ";
+	  cin >> n;
+	  cout << endl << "End: ";
+	  cin >> m;
+	  char letter;
+	  bool areValidNumbers = n >= 0 && n <= 255 && m >= 0 && m <= 255;
+
+	  if (areValidNumbers) {
+
+		  for (int i = n; i <= m; i++) {
+		  letter = i;
+		  cout << i << " --> " << letter << endl;
+			  }
+			}
+	  else {
+		  cout << "Invalid codes!";
+	  }*/
+
+	  //zad3
+	  /*int n;
+	  cin >> n;
+	  int y = 0;
+
+	  for (int x = 1; x < n; x++) {
+		  y = n - x;
+		  cout << "x = " << x << ", y = " << y << endl;
+	  }*/
+
+	  //zad4
+	  /*long long n;
+	  cin >> n;
+	  int sum = 0;
+	  int currentDigit = 0;
+
+	  while (n>0) {
+		   currentDigit = n % 10;
+		  sum += currentDigit;
+		  n /= 10;
+	  }
+
+	  while (sum >= 10) {
+		  int newNumber = sum;
+		  sum = 0;
+		  currentDigit = 0;
+
+		  while(newNumber > 0){
+			  currentDigit = newNumber % 10;
+			  sum += currentDigit;
+			  newNumber /= 10;
+		  }
+	  }
+	  cout << sum;*/
+
+	//zad5
+	/*int n;
+	cin >> n;
+	int sqr = 1;
+	if (n > 0) {
+		sqr = n * n;
+
+		while (sqr > 0) {
+			int currentDigit = sqr % 10;
+			sqr /= 10;
+
+			if (currentDigit != 0) {
+				cout << currentDigit;
+			}
+			else {
+				continue;
+			}
+		}
+	 }
+	 else {
+	 	cout << "Invalid number.";
+	 }*/
+
+	//zad6
+	/*int m, n;
+	cin >> m >> n;
+	int currentNumber = 1;
+	bool arePositive = m > 0 && n > 0;
+
+	if (arePositive) {
+         for (int i = 0; i < m; i++) {
+         
+         		for (int j = 1; j <= n; j++) {
+         			cout << currentNumber << " ";
+         			currentNumber ++;
+         		}
+         		cout << endl;
+         	}
+	}
+	else {
```

Comment:
make it if(!arePositive) cout <<"invalid output"
its not wrong but looks better